backend/pyproject.toml

```diff
@@ -0,0 +1,157 @@
+[tool.isort]
+py_version=312
+profile="black"
+line_length = 88
+lines_after_imports = 2
+
+sections=["FUTURE","STDLIB","THIRDPARTY","FIRSTPARTY","LOCALFOLDER"]
+
+order_by_type = false
+combine_as_imports = true
+force_grid_wrap = 3
+force_sort_within_sections = true
+include_trailing_comma = true
+use_parentheses = true
+
+
+[tool.ruff]
+exclude = [
+    ".bzr",
+    ".direnv",
+    ".eggs",
+    ".git",
+    ".git-rewrite",
+    ".hg",
+    ".mypy_cache",
+    ".nox",
+    ".pants.d",
+    ".pytype",
+    ".ruff_cache",
+    ".svn",
+    ".tox",
+    ".venv",
+    "__pypackages__",
+    "_build",
+    "buck-out",
+    "build",
+    "dist",
+    "node_modules",
+    "venv",
+    "setup.py",
+]
+
+# Same as Black.
+indent-width = 4
+
+target-version = "py312"
+
+[tool.ruff.format]
+# Like Black, use double quotes for strings.
+quote-style = "double"
+
+# Like Black, indent with spaces, rather than tabs.
+indent-style = "space"
+
+# Like Black, respect magic trailing commas.
+skip-magic-trailing-comma = false
+
+# Like Black, automatically detect the appropriate line ending.
+line-ending = "auto"
+
+# Enable auto-formatting of code examples in docstrings. Markdown,
+# reStructuredText code/literal blocks and doctests are all supported.
+#
+# This is currently disabled by default, but it is planned for this
+# to be opt-out in the future.
+docstring-code-format = true
+
+# Set the line length limit used when formatting code snippets in
+# docstrings.
+#
+# This only has an effect when the `docstring-code-format` setting is
+# enabled.
+docstring-code-line-length = "dynamic"
+
+preview = true
+
+[tool.ruff.lint]
+preview = true
+ignore = [
```

Comment:
here is the list of ruff violations currently ignored in this PR,
Ideally these would be addressed in follow-up PRs instead of ignored long term